src/org/apache/camel/idea/CamelContributor.java

```diff
@@ -117,4 +117,55 @@ public void addCompletions(@NotNull CompletionParameters parameters,
         }
     }
 
+    /**
+     * Parse the PSI text {@link CompletionUtil#DUMMY_IDENTIFIER} and " character and remove them
+     * @param parameters - completion parameter to parse
+     * @return new string stripped for any {@link CompletionUtil#DUMMY_IDENTIFIER} and " character
+     */
+    @NotNull
+    private static String parsePsiElement(@NotNull CompletionParameters parameters) {
+        String val = parameters.getPosition().getText();
+        int hackIndex = val.indexOf(CompletionUtil.DUMMY_IDENTIFIER);
+        if (hackIndex == -1) {
+            hackIndex = val.indexOf(CompletionUtil.DUMMY_IDENTIFIER_TRIMMED);
+        }
+        if (hackIndex > -1) {
+            val = val.substring(0, hackIndex);
+        }
+
+        if (val.startsWith("\"")) {
+            val = val.substring(1);
+        }
+        if (val.endsWith("\"")) {
+            val = val.substring(0, val.length() - 1);
+        }
+        return val;
+    }
+
+    private static void propertyPlaceholdersSmartCompletion(@NotNull CompletionParameters parameters, @NotNull CompletionResultSet resultSet) {
+        VfsUtil.processFilesRecursively(parameters.getOriginalFile().getManager().getProject().getBaseDir(), virtualFile -> {
+            if (virtualFile.getName().endsWith(".properties")) {
+                File file = new File(virtualFile.getPath());
+                Properties properties = new Properties();
+                try {
+                    properties.load(Files.newInputStream(file.toPath()));
+                } catch (IOException e) {
+                }//TODO : log a warning, but for now we ignore it and continue.
+
+                properties.forEach((key, value) -> {
+                            String keyStr = (String) key;
+                            boolean noneMatch = IGNORE_PROPERTIES.stream().noneMatch(s -> keyStr.startsWith(s));
```

Comment:
yes, which mean none of the keys that is matching the filter list. 

The filter is only on key level, so any non ignore keys in a property file will still be included.